board.js

```diff
@@ -0,0 +1,59 @@
+var Board = function(id, tiles, num_rows, num_columns){
+  this.id = id;
+  this.tiles = tiles;
+  this.size = tiles.length;
+  this.num_rows = num_rows;
+  this.num_columns = num_columns;
+};
+
+Board.prototype = {
+  
+  // returns a list of all Tiles that have matched with another
+  allMatchedTiles: function(){
+    return this.tiles.filter( function(tile){ return tile.matched; });
+  },
+
+  // returns a list of all Tiles that are activated
+  allActivatedTiles: function() {
+    return this.tiles.filter( function(tile){ return tile.activated; });
+  },
+
+  // returns all tiles
+  allTiles: function() {
+    return this.tiles;
+  },
+
+  // get the tile with this id
+  getTile: function(tile_id){
+    return this.tiles.find( function(tile){ return tile.id == tile_id; }); 
+  },
+
+  // clears the board - removes table
+  clear: function(){
+    $("#"+this.id).html("");    
```

Comment:
FYI, often you'll see that classes store their corresponding element as a property in the constructor

```javascript
this.$el = $("#"+this.id);
```

so that it doesn't have to be re-retrieved from the DOM repeatedly, which is a relatively slow operation.